testcases/open_posix_testsuite/include/proc.h

```diff
@@ -56,7 +63,12 @@ int tst_process_state_wait3(pid_t pid, const char state,
 int tst_process_state_wait3(pid_t pid, const char state,
 	unsigned long maxwait_ms)
 {
-	usleep(maxwait_ms * 1000);
+	struct timespec maxwait_ts;
+
+	maxwait_ts.tv_sec = 0;
+	maxwait_ts.tv_nsec = maxwait_ms * 1000000;
```

Comment:
maxwait_ms can be higher than 1000ms, so this can "overflow" tv_nsec.

Reply:
This would have been an issue previously, in a different manner. In particular, `usleep(3)` on linux requires types of value `useconds_t`, which is a `uint32_t` value.

Reply:
But to reach that problem with usleep() you'd need 1000x higher values. uint32_t can hold 5000*1000.

nanosecond version can overflow in 2 different ways. One is you can overflow with 32bit types, the other one is you can't pass more than 1000ms to tv_nsec. You are guaranteed to hit second one with sem_post/8-1.c, which is passing 2000ms:
```
nanosleep({0, 2000000000}, NULL)        = -1 EINVAL (Invalid argument)
```